core/sbt-cloudflow/src/main/scala/cloudflow/sbt/CloudflowBasePlugin.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cloudflow.sbt
+
+import java.nio.charset.StandardCharsets._
+import java.util.Base64
+import java.util.zip.Deflater
+import java.io.ByteArrayOutputStream
+
+import com.typesafe.config._
+
+import sbt._
+import sbt.Keys._
+import sbtdocker.DockerKeys._
+import com.typesafe.sbt.packager.archetypes._
+import spray.json._
+
+import cloudflow.sbt.CloudflowKeys._
+import cloudflow.blueprint.StreamletDescriptorFormat._
+import cloudflow.blueprint.StreamletDescriptor
+
+abstract class CloudflowBasePlugin extends AutoPlugin {
```

Comment:
Could you add some documentation for the use of this class and what it does? It would help the readability of the code.

Reply:
done ..